src/components/Timer.jsx

```diff
@@ -1,39 +1,72 @@
 import React, { Component } from 'react';
-import formatTime from '../js/helper';
-
+import { formatTime, alertMessage } from '../js/helper';
+import uuid from 'uuid-v4';
+// 25 min/ 10 min  is default setting
 class Timer extends Component {
 	state = {
+		type: '',
+		title: '',
 		startTime: '',
 		duration: 25,
 		remained: 25 * 60,
-		isTimerRunning: false
+		breakTime: 10 * 60,
+		isTimerRunning: false,
+		isBreakRunning: false,
+	};
+	// breakTime will be passes as props from Dashboard in future
+	componentDidUpdate() {
+		if (!this.state.remained && this.state.isTimerRunning) {
+			this.handleComplete();
+		}
+		if (this.state.breakTime < 0) {
+			this.handleBreakOver();
+		}
+	}
+	countDown = () => {
+		const remainedSeconds = this.state.remained - 1;
+		this.setState({
+			remained: remainedSeconds
+		});
+	};
+	breakTimeCountDown = () => {
+		const breakTime = this.state.breakTime - 1;
+		this.setState({ breakTime });
 	};
-
 	resetTimer = newDuration => {
 		clearInterval(this.countDownID);
 		this.setState({
+			type: '',
+			title: '',
 			startTime: '',
 			duration: newDuration,
+			breakTime: 10 * 60,
 			remained: newDuration * 60,
-			isTimerRunning: false
+			isTimerRunning: false,
 		});
 	};
-
-	countDown = () => {
-		const remainedSeconds = this.state.remained - 1;
+	handleRecordSubmit = () => {
+		const type = this.state.type;
```

Comment:
Where do we store the `type` and `title`? When I finished my 25 minutes timer, I could see the time and the timer id but the type and title were empty strings

Reply:
it becomes empty because of ```this.resetTimer()``` function but the data has been sent to Dashboard!